File: Platform/codesignal/Intro/24.Minesweeper/solution.go
package main

import "fmt"

func solution(m [][]bool) [][]int {

	row := len(m)
	col := len(m[0])

	var result [][]int
	for i := 0; i < row; i++ {
		var row []int
		for j := 0; j < col; j++ {
			row = append(row, 0)
		}
		result = append(result, row)
	}

	for r := 0; r < row; r++ {
		for c := 0; c < col; c++ {
			n := 0
			if (r-1) >= 0 && (c-1) >= 0 && m[r-1][c-1] {
				n++
			}
			if (r-1) >= 0 && m[r-1][c] {
				n++
			}
			if (r-1) >= 0 && (c+1) < col && m[r-1][c+1] {
				n++
			}
			if (c+1) < col && m[r][c+1] {
				n++
			}
			if (r+1) < row && (c+1) < col && m[r+1][c+1] {
				n++
			}
			if (r+1) < row && m[r+1][c] {
				n++
			}
			if (r+1) < row && (c-1) >= 0 && m[r+1][c-1] {
				n++
			}
			if (c-1) >= 0 && m[r][c-1] {
				n++
			}
			result[r][c] = n
		}
	}
	return result
}

func main() {

	m := [][]bool{
		{true, false, false},
		{false, true, false},
		{false, false, false},
	}

	fmt.Println(solution(m))

}

File: Platform/codesignal/Intro/24.Minesweeper/solution.js
function solution(m){
  const [row, col] = [m.length, m[0].length]
  let result = Array(row).fill().map(() => Array(col).fill(0))

  for (let r=0; r<row; r++) {
    for (let c=0; c<col; c++) {
      let num = 0;
      if (r-1 >=0 && c-1 >=0 && m[r-1][c-1]) num++
      if (r-1 >=0 && m[r-1][c]) num++
      if (r-1 >=0 && c+1 < col && m[r-1][c+1]) num++
      if (c+1 < col && m[r][c+1]) num++
      if (r+1 < row && c+1 < col && m[r+1][c+1]) num++
      if (r+1 < row && m[r+1][c]) num++
      if (r+1 < row && c-1 >=0 && m[r+1][c-1]) num++
      if (c-1 >=0 && m[r][c-1]) num++
      result[r][c] = num
    }
  }
  return result
}

console.log(solution([[true,false,false], 
                      [false,true,false], 
                      [false,false,false]]));